### PR TITLE
Bool file sanitization

### PR DESCRIPTION
**Pull request by zyga**
Note: this pull request depends on #322 

To enhance the security of the bool-file capability it must not be allowed to create an arbitrary bool-file (e.g. one pointing to /dev/sda or /dev/mem). This is achieved with simple validation against a set of regular expressions describing allowed values.

Currently two kinds of bool file can be created. They can point to:
- The value of a sysfs GPIO 
- The brightness of a sysfs LED

Additional values can be added as we discover their utility.

### CODE REVIEW

**Comment by zyga** [comment]
/me goes to fix daemon code tests to use allowed paths.

**Comment by chipaca** [comment]
Dear mo, retest this please.

**Comment by chipaca** [comment]
How relevant to what we're building is the fact that `brightness` is not always a `0/1` boolean? E.g.,
```sh-session
$ tail /sys/class/leds/dell::kbd_backlight/*brightness
==> /sys/class/leds/dell::kbd_backlight/brightness <==
3

==> /sys/class/leds/dell::kbd_backlight/max_brightness <==
10
```

File: caps/types.go [hunk 1]
```diff
@@ -53,6 +54,13 @@ func (t *BoolFileType) Name() string {
 	return "bool-file"
 }
 
+var boolFileAllowedPathPatterns = []*regexp.Regexp{
+	// The brightness of standard LED class device
+	regexp.MustCompile("^/sys/class/leds/[^/]+/brightness$"),
```
Comment (niemeyer) [inline caps/types.go]: Given @chipaca's feedback, doesn't look like this is right.

File: caps/types.go [hunk 2]
```diff
@@ -63,7 +71,16 @@ func (t *BoolFileType) Sanitize(c *Capability) error {
 	if path == "" {
 		return fmt.Errorf("bool-file must contain the path attribute")
 	}
-	// TODO: validate the path against a regular expression
+	valid := false
+	for _, pattern := range boolFileAllowedPathPatterns {
+		if pattern.MatchString(path) {
+			valid = true
```
Comment (niemeyer) [inline caps/types.go]: `return true`, and then simplify the rest of the logic accordingly?

**Comment by niemeyer** [comment]
LGTM considering these points.. I'd drop brightness altogether for now.

File: caps/types.go (hunk 1 above)
Comment (zyga) [inline caps/types.go]: I'd like to keep it for two reasons:
 - it is useful for common keyboard LEDs (caps lock, scroll lock, etc) and for testing on x86
 - vast majority of LEDs have only two states 0 and !0  (no PWM support)

On a given dell laptop, the gadget snap can suppress that LED capability if it doesn't behave correctly. In every other case LEDs are a valid _candidate_ and thus should be allowed.

File: caps/types.go (hunk 2 above)
Comment (zyga) [inline caps/types.go]: Done

File: caps/types.go (hunk 1 above)
Comment (niemeyer) [inline caps/types.go]: Please just drop it. This is a very simple test capability that we can improve at will later. This branch could already be merged by now.

Comment (niemeyer) [inline caps/types.go]: Sorry, I take that back. Yes, these cases indeed work with 0 or 1, so we can instead blacklist whatever we don't want later.

**Comment by niemeyer** [comment]
And it's merged by now! ;)

**Pull request merged**